lib/datadog/statsd/forwarder.rb

```diff
@@ -24,6 +24,11 @@ def initialize(
       )
         @transport_type = socket_path.nil? ? :udp : :uds
 
+        # in order to add the telemetry metrics only in the main process
+        # note that it also means that if the main process is not submitting metrics,
+        # we won't send any telemetry data
```

Comment:
I'm curious -- what's the use and expected semantics for this telemetry?

If I'm reading this right, up until now (and even in the 4.x series), telemetry was always sent by all forks, which meant that we were actually already suffering from the "duplicated data" issue with telemetry as well.

This change makes telemetry only cover one of the processes (the "original"). Does this mean that we'll miss some of that data that previously came from the forked children? Is that expected?

Reply:
This was a wrong assumption (on top of sending wrong data if a fork happens).
Since we are using counters for the telemetry, multiple threads can send the data.